docs/src/main/asciidoc/hibernate-orm-panache-guide.adoc

```diff
@@ -381,6 +381,8 @@ Make sure to wrap methods modifying your database (e.g. `entity.persist()`) with
 CDI bean method `@Transactional` will do that for you and make that method a transaction boundary. We recommend doing 
 so at your application entry point boundaries like your REST endpoint controllers.
 
+Be aware that JPA may send the changes you done on your entities only at transaction boundary, this means that those changes (insert/update/delete queries) will be sent to the database after the execution of your method so you will not be able to catch any `PersistenceException` in your code if you need to (for example to send an event somewhere if you cannot commit to the database). Panache provides `flush()` and `persistAndFlush()` methods to force a flush on the underlying `EntityManager`. Calling one of these methods will send all pending changes (of all entities) to the database (but will not commit the transaction).
```

Comment:
s/you done/you did/

@emmanuelbernard @gsmet WDYT, this sounds like advice that is ORM-related and not at all Panache-specific, no? Should we document this in the ORM guide and point to it here?

Reply:
Corrected.

My point of view is that if someone use the JPA way (so hibernate ORM), they should know about this ;) But using Panache abstract JPA away so it must be documented!

But I can add the same explaination on the Hibernate guide with an example showing the entityManager.flush() method ... but again it's standard JPA ...

Reply:
@loicmathieu Let me propose a different wording less end of the worldy.

JPA batches changes you make to entity and sends changes (it's called flush) at the end of the transaction or before a query.
This is usually a good thing as it's more efficient.
But if you want to check optimistic locking failures, do object validation right away or generally want to get immediate feedback, you can force the flush operation by calling `entity.flush()` or even use `entity.persistAndFlush()` to make it a single method call.
Remember, this is less efficient so don't abuse it.
And your transaction still has to be committed.

I removed the reference to the message as it's a bad example. DB can still fail to commit even when flush happened.

Reply:
I think it's important to keep the reference to the fact that with this we can catch PesistenceException.

@emmanuelbernard I would propose something like this:

JPA batches changes you make to entity and sends changes (it's called flush) at the end of the transaction or before a query.
This is usually a good thing as it's more efficient.
But if you want to check optimistic locking failures, do object validation right away or generally want to get immediate feedback, you can force the flush operation by calling entity.flush() or even use entity.persistAndFlush() to make it a single method call. This will allow you to catch any `PersistenceException` that could occurs when JPA send those changes to the database.
Remember, this is less efficient so don't abuse it.
And your transaction still has to be committed.